lib/generators/suspenders/email_generator.rb

```diff
@@ -0,0 +1,44 @@
+module Suspenders
+  module Generators
+    class EmailGenerator < Rails::Generators::Base
+      source_root File.expand_path("../../templates/email", __FILE__)
+      desc <<~TEXT
+        Intercepts emails in non-production environments by setting `INTERCEPTOR_ADDRESSES`.
+
+        Configures `default_url_options` in `test` and `development`.
+      TEXT
+
+      def create_email_interceptor
+        copy_file "email_interceptor.rb", "app/mailers/email_interceptor.rb"
+      end
+
+      def create_email_interceptor_initializer
+        initializer "email_interceptor.rb", <<~RUBY
+          Rails.application.configure do
+            if ENV["INTERCEPTOR_ADDRESSES"].present?
+              config.action_mailer.interceptors = %w[EmailInterceptor]
+            end
+          end
+        RUBY
+      end
+
+      def configure_email_interceptor
+        environment do
+          %(
+            config.to_prepare do
+              EmailInterceptor.config.interceptor_addresses = ENV.fetch("INTERCEPTOR_ADDRESSES", [])
+            end
+          )
+        end
+      end
+
+      def configure_development_environment
+        environment %(config.action_mailer.default_url_options = { host: "localhost", port: 3000 }), env: "development"
+      end
+
+      def configure_test_environment
+        environment %(config.action_mailer.default_url_options = { host: "www.example.com" }), env: "test"
+      end
```

Comment:
These might be able to be removed if https://github.com/rails/rails/pull/51191 is merged.

Reply:
We can remove once there's a new release.